src/api/artifact/abstractor/resolver/chart/chart.go

```diff
@@ -9,3 +13,68 @@
 // limitations under the License.
 
 package chart
+
+import (
+	"context"
+	"encoding/json"
+	"github.com/goharbor/harbor/src/api/artifact/abstractor/blob"
+	resolv "github.com/goharbor/harbor/src/api/artifact/abstractor/resolver"
+	"github.com/goharbor/harbor/src/common/utils/log"
+	"github.com/goharbor/harbor/src/pkg/artifact"
+	"github.com/goharbor/harbor/src/pkg/repository"
+	v1 "github.com/opencontainers/image-spec/specs-go/v1"
+)
+
+const (
+	// ArtifactTypeChart defines the artifact type for helm chart
+	ArtifactTypeChart = "CHART"
+)
+
+func init() {
+	resolver := &resolver{
+		repoMgr:     repository.Mgr,
+		blobFetcher: blob.Fcher,
+	}
+	if err := resolv.Register(resolver, "application/vnd.cncf.helm.config.v1+json"); err != nil {
+		log.Errorf("failed to register resolver for artifact %s: %v", resolver.ArtifactType(), err)
+		return
+	}
+}
+
+type resolver struct {
+	repoMgr     repository.Manager
+	blobFetcher blob.Fetcher
+}
+
+func (r *resolver) ArtifactType() string {
+	return ArtifactTypeChart
+}
+
+func (r *resolver) Resolve(ctx context.Context, manifest []byte, artifact *artifact.Artifact) error {
+	repository, err := r.repoMgr.Get(ctx, artifact.RepositoryID)
+	if err != nil {
+		return err
+	}
+	m := &v1.Manifest{}
+	if err := json.Unmarshal(manifest, m); err != nil {
+		return err
+	}
+	digest := m.Config.Digest.String()
+	layer, err := r.blobFetcher.FetchLayer(repository.Name, digest)
+	if err != nil {
+		return err
+	}
+	// TODO should we abstract all values?
```

Comment:
I think we should extract the data similar to the previous chart info block and put the all the left ones into the extra properties which are rendered as k-v pairs to the users

Reply:
Seems most of the info abstracted from the yaml can only be stored in the extra properties